src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.HillClimbing.cs

```diff
@@ -0,0 +1,414 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        
+        private class HillClimbing
+        {
+            private struct Complex
+            {
+                public Complex(double real, double imaginary)
+                {
+                    Real = real;
+                    Imaginary = imaginary;
+                }
+
+                public double Imaginary { get; }
+                public double Real { get; }
+
+                public static Complex operator*(double scalar, Complex complex) => new Complex(scalar * complex.Real, scalar * complex.Imaginary);
```

Comment:
>double scalar, Complex complex [](start = 48, length = 30)

Since multiply is commutative, it would be good to provide an overload (even if it's not used) that takes parameters in the opposite order and calls into this overload